source/Halibut/DataStream.cs

```diff
@@ -131,9 +131,9 @@ public void CopyAndReportProgress(Stream destination)
                 while (count > 0)
                 {
                     Swap(ref readBuffer, ref writeBuffer);
-                    var asyncResult = destination.WriteAsync(writeBuffer, 0, count);
+                    var asyncResult = destination.BeginWrite(writeBuffer, 0, count, null, null);
                     count = source.Read(readBuffer, 0, BufferSize);
-                    asyncResult.GetAwaiter().GetResult();
+                    destination.EndWrite(asyncResult);
```

Comment:
This was how it was before the .NET Core conversion.

Reply:
BeginWrite/EndWrite (and similar on other APIs) weren't in .net core 1. Good to have them back.